scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/attestation/GramineQuoteGeneratorImpl.scala

```diff
@@ -17,14 +17,57 @@
 package com.intel.analytics.bigdl.ppml.attestation
 
 import org.apache.logging.log4j.LogManager
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
 
 class GramineQuoteGeneratorImpl extends QuoteGenerator {
 
   val logger = LogManager.getLogger(getClass)
 
-  override def getQuote(): Unit = {
-    // TODO Gramine Quote https://graphene.readthedocs.io/en/latest/attestation.html
-    logger.debug("Write user report to /dev/attestation/user_report_data")
-    logger.debug("Read quote from /dev/attestation/quote")
+  val userReportPath = "/dev/attestation/user_report_data"
+  val quotePath = "/dev/attestation/quote"
+
+  @throws[Exception]
+  override def getQuote(userReportData: Array[Byte]): Array[Byte] = {
+
+    if (userReportData.length == 0 || userReportData.length > 32) {
+      logger.error("Incorrect userReport size!")
+      throw new Exception("Incorrect userReportData size!")
+    }
+
+    try {
+      // write userReport 
+      val out = new BufferedOutputStream(new FileOutputStream(userReportPath))
+      out.write(userReportData)
+      out.close()
+    } catch {
+      case e: Exception => {
+        logger.error(s"Failed to write user report, ${e}")
+        e.printStackTrace()
+      }
+    }
+
+    var quote: Array[Byte] = null
```

Comment:
I think quote can be val

Reply:
Val type for quote will raise a compilation error later when we try to assign a new value to quote since val in scala is const; hence we use var here.

Reply:
can use expression as value like below:
val quote = {
 ........
 new Array[Byte](quoteFile.length.toInt)
}

Reply:
Quote is now assigned as a local val, and is returned when workflow is successful.